Assets/Scripts/Models/Power/Connection.cs

```diff
@@ -100,6 +114,11 @@ public void WriteXml(XmlWriter writer)
             writer.WriteAttributeString(accumulatedPowerAttributeName, AccumulatedPower.ToString(CultureInfo.InvariantCulture));
         }
 
+        public object Clone()
```

Comment:
Why not make the return value a Connection, since that is exactly what this returns. You won't need to cast it after that change.